### ISSUE TITLE
Featureplot not plotting if one of the variables is missing from the seurat object

### ISSUE THREAD

**Issue by Prathyusha-konda** [samuel-marsh/scCustomize#64]
Feature plot quits plotting if one of the variables is missing instead of plotting existing variables.

Error in FetchData.Seurat(object = seurat_object, vars = features[i],  : 
  None of the requested variables were found: CD279

> sessionInfo()
R version 4.2.1 (2022-06-23)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.7 LTS

Matrix products: default
BLAS:   /usr/lib/atlas-base/atlas/libblas.so.3.0
LAPACK: /opt/R/4.2.1/lib/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readr_2.1.3        scCustomize_0.7.0  RColorBrewer_1.1-3 glmGamPoi_1.9.2    pheatmap_1.0.12    cowplot_1.1.1     
 [7] patchwork_1.1.2    sp_1.5-0           SeuratObject_4.1.2 Seurat_4.2.0       dplyr_1.0.10

**Comment by samuel-marsh** [comment]
Hi,

Could you please post the full code that you are running which causes the error?  Also, if you update scCustomize to the develop branch does that resolve the issue?

```
devtools::install_github(repo = "samuel-marsh/scCustomize", ref = "develop")
```

When I run the following code (using dev branch) it seems to work ok.

```
library(tidyverse)
library(scCustomize)
library(Seurat)

pbmc <- pbmc3k.SeuratData::pbmc3k.final
pbmc <- UpdateSeuratObject(pbmc)

FeaturePlot_scCustom(seurat_object = pbmc, c("FOS", "fdskjf", "CD3E"))

Warning message:
In FetchData.Seurat(object = object, vars = c(dims, "ident", features),  :
  The following requested variables were not found: fdskjf
```

![image](https://user-images.githubusercontent.com/38284410/196226270-bafe6546-f4f6-4be5-ac37-fff4ff33a31a.png)

**Comment by Prathyusha-konda** [comment]
Hi Sam, thanks for the quick response. So it looks like featureplot doesn't plot when you include 'split.by' option and have a missing variable in the list. Without split.by, it works fine even with missing variable.

Example code to reproduce using seurat's ifnb data

# install dataset
InstallData("ifnb")

# load dataset
LoadData("ifnb")

# split the dataset into a list of two seurat objects (stim and CTRL)
ifnb.list <- SplitObject(ifnb, split.by = "stim")

ctrl <- ifnb.list[["CTRL"]]
stim <- ifnb.list[["STIM"]]

# normalize and run dimensionality reduction on control dataset
ctrl <- SCTransform(ctrl, vst.flavor = "v2", verbose = FALSE) %>%
    RunPCA(npcs = 30, verbose = FALSE)

stim <- SCTransform(stim, vst.flavor = "v2", verbose = FALSE) %>%
    RunPCA(npcs = 30, verbose = FALSE)

ifnb.list <- list(ctrl = ctrl, stim = stim)
features <- SelectIntegrationFeatures(object.list = ifnb.list, nfeatures = 3000)
ifnb.list <- PrepSCTIntegration(object.list = ifnb.list, anchor.features = features)

immune.anchors <- FindIntegrationAnchors(object.list = ifnb.list, normalization.method = "SCT",
    anchor.features = features)
immune.combined.sct <- IntegrateData(anchorset = immune.anchors, normalization.method = "SCT")

immune.combined.sct <- RunPCA(immune.combined.sct, verbose = FALSE)
immune.combined.sct <- RunUMAP(immune.combined.sct, reduction = "pca", dims = 1:30, verbose = FALSE)
immune.combined.sct <- FindNeighbors(immune.combined.sct, reduction = "pca", dims = 1:30)
immune.combined.sct <- FindClusters(immune.combined.sct, resolution = 0.3)

immune.combined.sct$celltype.stim <- paste(immune.combined.sct$seurat_annotations, immune.combined.sct$stim,
    sep = "_")
Idents(immune.combined.sct) <- "celltype.stim"

pdf("featureplot_test.pdf")
FeaturePlot_scCustom(immune.combined.sct, features = c("CD3D", "GNLY", "FLT1"))
dev.off()

Found the following features in more than one assay, excluding the default. We will not include these in the final data frame: FLT1Error in FetchData.Seurat(object = seurat_object, vars = features[i],  : 
  None of the requested variables were found: FLT1

**Comment by samuel-marsh** [comment]
Ok fix is live in the develop branch v(0.7.0.9951).  

If setting `split.by` the function will check for features against the assay features and meta data columns and return a warning of which features were not found and return plot with found features.

If you have any issues once updated please let me know and I'll reopen the issue.

Best,
Sam

**Comment by Prathyusha-konda** [comment]
> Ok fix is live in the develop branch v(0.7.0.9951).
> 
> If setting `split.by` the function will check for features against the assay features and meta data columns and return a warning of which features were not found and return plot with found features.
> 
> If you have any issues once updated please let me know and I'll reopen the issue.
> 
> Best, Sam

That's awesome! Thank you, Sam! Looking forward to the new release 👍 💯

**Comment by samuel-marsh** [comment]
glad to help!!! :)
